pkg/eks/api.go

```diff
@@ -300,7 +299,7 @@ func (c *ClusterProvider) GetCredentialsEnv() ([]string, error) {
 
 // checkAuth checks the AWS authentication
 func (c *ClusterProvider) checkAuth() error {
-	output, err := c.Provider.STSV2().GetCallerIdentity(context.TODO(), &stsv2.GetCallerIdentityInput{})
+	output, err := c.Provider.STS().GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
```

Comment:
We should propagate the context here.

Reply:
From where?

Reply:
I meant by adding a `Context` parameter to `checkAuth` and passing it from the top-most call.

Reply:
I don't necessarily agree with that. :) But that's not a hill I'll die on. :D

Reply:
I'd agree with @cPu1 , we've done that for other v2 migrations, not sure why we wouldn't here :thinking:

Reply:
Fine, like I said, I don't agree, but I don't care enough. :D

Reply:
you are the ones who will review it hahahaha

Reply:
Done :)

Reply:
Okay, it's not that bad actually. :D